src/pool/srv_cli.c

```diff
@@ -382,34 +382,42 @@ process_query_result(d_rank_list_t **enabled_ranks, d_rank_list_t **disabled_ran
 	rc = pool_map_create(map_buf, map_version, &map);
 	if (rc != 0) {
 		DL_ERROR(rc, DF_UUID ": failed to create local pool map", DP_UUID(pool_uuid));
-		D_GOTO(error, rc = rc);
+		D_GOTO(error, rc);
 	}
 
 	rc = pool_map_find_failed_tgts(map, NULL, &num_disabled);
 	if (rc != 0) {
 		DL_ERROR(rc, DF_UUID ": failed to get num disabled tgts", DP_UUID(pool_uuid));
-		D_GOTO(error, rc = rc);
+		D_GOTO(error, rc);
 	}
 
 	if ((pi_bits & DPI_ENGINES_ENABLED) != 0) {
-		D_ASSERT(enabled_ranks != NULL);
+		if (enabled_ranks == NULL) {
+			DL_ERROR(-DER_INVAL, DF_UUID ": query pool with invalid params",
```

Comment:
minor - might be nice to be a little more descriptive here. Something like "query pool requested enabled ranks, but ptr is NULL"?

Reply:
- [ ] Fix error message

Reply:
> * [x]  Fix error message

Fixed with commit 533bf932f